test/fuzz/xds_fuzz_test.go

```diff
@@ -9,24 +9,164 @@ import (
 	"strings"
 	"testing"
 
-	fuzz "github.com/AdaLogics/go-fuzz-headers"
-
 	"github.com/envoyproxy/gateway/internal/cmd/egctl"
 	"github.com/envoyproxy/gateway/internal/gatewayapi/resource"
 )
 
 func FuzzGatewayAPIToXDS(f *testing.F) {
+	// Add seed corpus for golang native fuzzing. OSS-Fuzz will not take these corpus into consideration.
+	// grpc route
+	f.Add([]byte(`apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: eg
+spec:
+  controllerName: gateway.envoyproxy.io/gatewayclass-controller
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: eg
+  namespace: default
+spec:
+  gatewayClassName: eg
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: GRPCRoute
+metadata:
+  name: backend
+  namespace: default
+spec:
+  parentRefs:
+    - name: eg
+      sectionName: grpc
+  hostnames:
+    - "www.grpc-example.com"
+  rules:
+    - matches:
+        - method:
+            service: com.example.Things
+            method: DoThing
+          headers:
+            - name: com.example.Header
+              value: foobar
+      backendRefs:
+        - name: provided-backend
+          port: 9000
+---
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: Backend
+metadata:
+  name: provided-backend
+  namespace: default
+spec:
+  endpoints:
+    - ip:
+        address: 0.0.0.0
+        port: 8000`))
+
+	// http route
+	f.Add([]byte(`apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
```

Comment:
the same gateway class and gateway is being added multiple times, do you want to split it up as
1 step adds - GWC and gateways
other steps install routes

Reply:
I see, yeah we can try to minimize duplication. I will update shortly

Reply:
Updated the seed definition, In Go fuzzing `(f.Fuzz(...))` must operate per input and has no memory of previous f.Add() calls once the fuzzer is running. So each seed has to be complete itself.